# Leqi Zhong final project notebook 2

In notebook 2, I'll explore the following three questions:
1. The change of students applying for free meals over the past five years.
2. Ratio of free meals to FRPM.
3. The change in the number of 100% free meals schools.

In [2]:
import pandas as pd
import altair as alt

In [3]:
data_2122_clean = pd.read_csv('data_2122_clean.csv')

In [4]:
data_2021_clean = pd.read_csv('data_2021_clean.csv')

In [5]:
data_1920_clean = pd.read_csv('data_1920_clean.csv')

In [6]:
data_1819_clean = pd.read_csv('data_1819_clean.csv')

In [7]:
data_1718_clean = pd.read_csv('data_1718_clean.csv')

## Students applying for free meals over the past five years

In [8]:
data_1819_clean['1819_Free_Meal_Count'].mean()

307.63783269961976

In [9]:
data_2122_clean['2122_Free_Meal_Count'].mean()

284.40528509187345

Hmmm, and I realized calculating absolute values doesn't make much sense (because each school has a different number of students), so let's look at the median percentage.

In [10]:
data_1718_clean['1718_Free_Meal_Percent'].mean()

0.5537593317422433

In [11]:
data_1819_clean['1819_Free_Meal_Percent'].mean()

0.5464433460076046

In [12]:
data_1920_clean['1920_Free_Meal_Percent'].mean()

0.5408470543806646

In [13]:
data_2021_clean['2021_Free_Meal_Percent'].mean()

0.5451900066369584

In [14]:
data_2122_clean['2122_Free_Meal_Percent'].mean()

0.5304205341920818

That's interesting! I thought more students would apply for free meals after the pandemic because of the economic downturn. Maybe I should also look at other columns?

In [15]:
data_1718_clean['1718_FRPM_Count'].mean()

356.9782338902148

In [16]:
data_1819_clean['1819_FRPM_Count'].mean()

349.3468631178707

In [17]:
data_2021_clean['2021_FRPM_Count'].mean()

335.05499194083626

In [18]:
data_2122_clean['2122_FRPM_Count'].mean()

322.4637241901875

How about the percentage?

In [19]:
data_1819_clean['1819_FRPM_Percent'].mean()

0.6114997148288972

In [20]:
data_1920_clean['1920_FRPM_Percent'].mean()

0.60825708081571

In [21]:
data_2021_clean['2021_FRPM_Percent'].mean()

0.6033516639802787

In [22]:
data_2122_clean['2122_FRPM_Percent'].mean()

0.5917071415040728

The number has been decreasing. Could it be the problem of enrollment?

In [23]:
data_1718_clean['1718_Enrollment'].sum()

6220826

In [24]:
data_1819_clean['1819_Enrollment'].sum()

6186628

In [25]:
data_1920_clean['1920_Enrollment'].sum()

6163338

In [26]:
data_2021_clean['2021_Enrollment'].sum()

6002523

In [27]:
data_2122_clean['2122_Enrollment'].sum()

5892240

Okay the enrollments have also been declining, see if we can match the trend with the free meal count.

In [28]:
data_1718_clean['1718_Free_Meal_Count'].sum()

3311699

In [29]:
data_1819_clean['1819_Free_Meal_Count'].sum()

3236350

In [30]:
data_1920_clean['1920_Free_Meal_Count'].sum()

3197492

In [31]:
data_2021_clean['2021_Free_Meal_Count'].sum()

3148051

In [32]:
data_2122_clean['2122_Free_Meal_Count'].sum()

3002751

Okay, though it's surprising, but we can draw conclusions from the above calculations:
1. Fewer students enrolled in schools after the pandemic.
2. Fewer students apply for free meals after the pandemic.

## Ratio of free meals to FRPM

There are two indicators, Free Meals and Free or Reduced-Price Meals (FRPM), I'm also curious about the percentage of free meals in FRPM.

I'm now creating a new column to show the difference for the 2122 data.

In [33]:
data_2122_clean['2122_Reduced-Price_Meals_Count'] = data_2122_clean['2122_FRPM_Count'] - data_2122_clean['2122_Free_Meal_Count']
data_2122_clean

Academic Year County Name                       District Name  \
0         2021-2022     Alameda  Alameda County Office of Education   
1         2021-2022     Alameda  Alameda County Office of Education   
2         2021-2022     Alameda  Alameda County Office of Education   
3         2021-2022     Alameda  Alameda County Office of Education   
4         2021-2022     Alameda  Alameda County Office of Education   
...             ...         ...                                 ...   
10553     2021-2022        Yuba                Wheatland Union High   
10554     2021-2022        Yuba                Wheatland Union High   
10555     2021-2022        Yuba                Wheatland Union High   
10556     2021-2022        Yuba                Wheatland Union High   
10557     2021-2022        Yuba                Wheatland Union High   

                                   School Name  \
0                     Alameda County Community   
1           Alameda County Juvenile Hall/Court   
2                       Alternatives in Action   
3                    Aurum Preparatory Academy   
4      Community School for Creative Education   
...                                        ...   
10553                          District Office   
10554                         Edward P. Duplex   
10555          Nonpublic, Nonsectarian Schools   
10556             Wheatland Community Day High   
10557                     Wheatland Union High   

                          School Type         Educational_Type Low Grade  \
0                    County Community  County Community School         K   
1              Juvenile Court Schools    Juvenile Court School         K   
2               High Schools (Public)              Traditional         9   
3        Junior High Schools (Public)              Traditional         6   
4         Elementary Schools (Public)              Traditional         K   
...                               ...                      ...       ...   
10553                             NaN                      NaN         7   
10554       Continuation High Schools      Continuation School         9   
10555                             NaN                      NaN       NaN   
10556  District Community Day Schools     Community Day School         7   
10557           High Schools (Public)              Traditional         9   

      High Grade  2122_Enrollment  2122_Free_Meal_Count  \
0             12               57                    45   
1             12               64                    64   
2             12              150                   113   
3              8              164                   135   
4              8              202                   162   
...          ...              ...                   ...   
10553         12                3                     3   
10554         12               45                    32   
10555        NaN                1                     0   
10556         12                5                     3   
10557         12             1012                   434   

       2122_Free_Meal_Percent  2122_FRPM_Count  2122_FRPM_Percent  \
0                       0.789               47              0.825   
1                       1.000               64              1.000   
2                       0.753              134              0.893   
3                       0.823              148              0.902   
4                       0.802              177              0.876   
...                       ...              ...                ...   
10553                   1.000                3              1.000   
10554                   0.711               45              1.000   
10555                   0.000                0              0.000   
10556                   0.600                4              0.800   
10557                   0.429              750              0.741   

       2122_Reduced-Price_Meals_Count  
0                                   2  
1                    

Oaky, Let's do the same for other years'.

In [34]:
data_2021_clean['2021_Reduced-Price_Meals_Count'] = data_2021_clean['2021_FRPM_Count'] - data_2021_clean['2021_Free_Meal_Count']
data_2021_clean

Academic Year County Name                       District Name  \
0         2020-2021     Alameda  Alameda County Office of Education   
1         2020-2021     Alameda  Alameda County Office of Education   
2         2020-2021     Alameda  Alameda County Office of Education   
3         2020-2021     Alameda  Alameda County Office of Education   
4         2020-2021     Alameda  Alameda County Office of Education   
...             ...         ...                                 ...   
10542     2020-2021        Yuba                Wheatland Union High   
10543     2020-2021        Yuba                Wheatland Union High   
10544     2020-2021        Yuba                Wheatland Union High   
10545     2020-2021      Alpine   Alpine County Office of Education   
10546     2020-2021      Sierra   Sierra County Office of Education   

                                   School Name  \
0                     Alameda County Community   
1           Alameda County Juvenile Hall/Court   
2                    Aurum Preparatory Academy   
3      Community School for Creative Education   
4         Connecting Waters Charter - East Bay   
...                                        ...   
10542                         Edward P. Duplex   
10543             Wheatland Community Day High   
10544                     Wheatland Union High   
10545                Alpine County Opportunity   
10546          Sierra County Special Education   

                          School Type          Educational_Type Low Grade  \
0                    County Community   County Community School         K   
1              Juvenile Court Schools     Juvenile Court School         K   
2        Junior High Schools (Public)               Traditional         6   
3         Elementary Schools (Public)               Traditional         K   
4               K-12 Schools (Public)               Traditional         K   
...                               ...                       ...       ...   
10542       Continuation High Schools       Continuation School         9   
10543  District Community Day Schools      Community Day School         7   
10544           High Schools (Public)               Traditional         9   
10545             Opportunity Schools        Opportunity School         1   
10546             Special Ed (Public)  Special Education School         K   

      High Grade  2021_Enrollment  2021_Free_Meal_Count  \
0             12              105                    78   
1             12               52                    52   
2              8              197                   136   
3              8              253                   169   
4             12              534                   122   
...          ...              ...                   ...   
10542         12               62                    39   
10543         12                3                     1   
10544         12              867                   329   
10545         12                0                     0   
10546         12                0                     0   

       2021_Free_Meal_Percent  2021_FRPM_Count  2021_FRPM_Percent  \
0                       0.743               80              0.762   
1                       1.000               52              1.000   
2                       0.690              146              0.741   
3                       0.668              176              0.696   
4                       0.228              146              0.273   
...                       ...              ...                ...   
10542                   0.629               47              0.758   
10543                   0.333                2              0.667   
10544                   0.379              422              0.487   
10545                   0.000                0              0.000   
10546                   0.000                0              0.000   

       2021_Reduced-Price_Meals_Count  
0                                   2  
1        

In [35]:
data_1920_clean['1920_Reduced-Price_Meals_Count'] = data_1920_clean['1920_FRPM_Count'] - data_1920_clean['1920_Free_Meal_Count']
data_1920_clean

Academic Year  County Name                       District Name  \
0         2019-2020      Alameda  Alameda County Office of Education   
1         2019-2020      Alameda  Alameda County Office of Education   
2         2019-2020      Alameda  Alameda County Office of Education   
3         2019-2020      Alameda  Alameda County Office of Education   
4         2019-2020      Alameda  Alameda County Office of Education   
...             ...          ...                                 ...   
10587     2019-2020         Yuba                Wheatland Union High   
10588     2019-2020         Yuba                Wheatland Union High   
10589     2019-2020  Los Angeles                              Lennox   
10590     2019-2020       Alpine   Alpine County Office of Education   
10591     2019-2020       Sierra   Sierra County Office of Education   

                                   School Name  \
0       Envision Academy for Arts & Technology   
1      Community School for Creative Education   
2                              Yu Ming Charter   
3                     Urban Montessori Charter   
4                                 Epic Charter   
...                                        ...   
10587                         Edward P. Duplex   
10588                     Wheatland Union High   
10589           Century Academy for Excellence   
10590                Alpine County Opportunity   
10591          Sierra County Special Education   

                                School Type          Educational_Type  \
0                     High Schools (Public)               Traditional   
1               Elementary Schools (Public)               Traditional   
2               Elementary Schools (Public)               Traditional   
3               Elementary Schools (Public)               Traditional   
4      Intermediate/Middle Schools (Public)               Traditional   
...                                     ...                       ...   
10587             Continuation High Schools       Continuation School   
10588                 High Schools (Public)               Traditional   
10589  Intermediate/Middle Schools (Public)               Traditional   
10590                   Opportunity Schools        Opportunity School   
10591                   Special Ed (Public)  Special Education School   

      Low Grade High Grade  1920_Enrollment  1920_Free_Meal_Count  \
0             6         12              412                   259   
1             K          8              267                   198   
2             K          8              487                    69   
3             K          8              393                   115   
4             6          8              246                   189   
...         ...        ...              ...                   ...   
10587         9         12               50                    27   
10588         9         12              814                   279   
10589         6          8                0                     0   
10590         1         12                0                     0   
10591         K         12                0                     0   

       1920_Free_Meal_Percent  1920_FRPM_Count  1920_FRPM_Percent  \
0                       0.629              295              0.716   
1                       0.742              228              0.854   
2                       0.142               77              0.158   
3                       0.293              132              0.336   
4                       0.768              199              0.809   
...                       ...              ...                ...   
10587                   0.540               30              0.600   
10588                   0.343              365              0.448   
10589                   0.000                0              0.000   
10590                   0.000                0              0.000   
10591                   0.000                0              0.000   

     

In [36]:
data_1819_clean['1819_Reduced-Price_Meals_Count'] = data_1819_clean['1819_FRPM_Count'] - data_1819_clean['1819_Free_Meal_Count']
data_1819_clean.head()

Academic Year County Name                       District Name  \
0     2018-2019     Alameda  Alameda County Office of Education   
1     2018-2019     Alameda  Alameda County Office of Education   
2     2018-2019     Alameda  Alameda County Office of Education   
3     2018-2019     Alameda  Alameda County Office of Education   
4     2018-2019     Alameda  Alameda County Office of Education   

                               School Name                  School Type  \
0   Envision Academy for Arts & Technology        High Schools (Public)   
1  Community School for Creative Education  Elementary Schools (Public)   
2                          Yu Ming Charter  Elementary Schools (Public)   
3                 Urban Montessori Charter  Elementary Schools (Public)   
4       Alameda County Juvenile Hall/Court       Juvenile Court Schools   

        Educational_Type Low Grade High Grade  1819_Enrollment  \
0            Traditional         9         12              385   
1            Traditional         K          8              241   
2            Traditional         K          8              445   
3            Traditional         K          8              432   
4  Juvenile Court School         K         12               53   

   1819_Free_Meal_Count  1819_Free_Meal_Percent  1819_FRPM_Count  \
0                   262                   0.681              306   
1                   118                   0.490              120   
2                    58                   0.130               64   
3                   113                   0.262              131   
4                    53                   1.000               53   

   1819_FRPM_Percent  1819_Reduced-Price_Meals_Count  
0              0.795                              44  
1              0.498                               2  
2              0.144                               6  
3              0.303                              18  
4              1.000                               0

In [37]:
data_1718_clean['1718_Reduced-Price_Meals_Count'] = data_1718_clean['1718_FRPM_Count'] - data_1718_clean['1718_Free_Meal_Count']
data_1718_clean.head()

Academic Year County Name                       District Name  \
0     2017-2018     Alameda  Alameda County Office of Education   
1     2017-2018     Alameda  Alameda County Office of Education   
2     2017-2018     Alameda  Alameda County Office of Education   
3     2017-2018     Alameda  Alameda County Office of Education   
4     2017-2018     Alameda  Alameda County Office of Education   

                               School Name                  School Type  \
0   Envision Academy for Arts & Technology        High Schools (Public)   
1  Community School for Creative Education  Elementary Schools (Public)   
2                          Yu Ming Charter  Elementary Schools (Public)   
3                 Urban Montessori Charter  Elementary Schools (Public)   
4       Alameda County Juvenile Hall/Court       Juvenile Court Schools   

        Educational_Type Low Grade High Grade  1718_Enrollment  \
0            Traditional         9         12              413   
1            Traditional         K          8              252   
2            Traditional         K          8              387   
3            Traditional         K          8              448   
4  Juvenile Court School         K         12               77   

   1718_Free_Meal_Count  1718_Free_Meal_Percent  1718_FRPM_Count  \
0                   265                   0.642              308   
1                   169                   0.671              180   
2                    49                   0.127               53   
3                   118                   0.263              127   
4                    77                   1.000               77   

   1718_FRPM_Percent  1718_Reduced-Price_Meals_Count  
0              0.746                              43  
1              0.714                              11  
2              0.137                               4  
3              0.283                               9  
4              1.000                               0

Now we can look at metrics in FRPM:

In [38]:
data_1718_clean['1718_Reduced-Price_Meals_Count'].sum()

427648

In [39]:
data_1819_clean['1819_Reduced-Price_Meals_Count'].sum()

438779

In [40]:
data_1920_clean['1920_Reduced-Price_Meals_Count'].sum()

457451

In [41]:
data_2021_clean['2021_Reduced-Price_Meals_Count'].sum()

385774

In [42]:
data_2122_clean['2122_Reduced-Price_Meals_Count'].sum()

401821

Good, we finally see some fluctuation.

In [43]:
data_1718_clean['1718_Reduced-Price_Meals_Count'].mean()

40.82558472553699

In [44]:
data_1819_clean['1819_Reduced-Price_Meals_Count'].mean()

41.70903041825095

In [45]:
data_2021_clean['2021_Reduced-Price_Meals_Count'].mean()

36.57665686925192

In [46]:
data_2122_clean['2122_Reduced-Price_Meals_Count'].mean()

38.058439098314075

So we have the third conclusion:

3. The number applying for reduced-price meals fluctuated, but the latest number is far below the pre-pandemic level.

## The change in the number of 100% free meals schools.

To achieve this, let me sort data according to free meals applying rate.

In [47]:
data_1718_clean.sort_values(by=['1718_Free_Meal_Percent'], ascending=False)

Academic Year     County Name                          District Name  \
10395     2017-2018            Yolo                     Washington Unified   
1815      2017-2018           Kings                    Reef-Sunset Unified   
4537      2017-2018           Modoc       Modoc County Office of Education   
6479      2017-2018      San Benito  San Benito County Office of Education   
6736      2017-2018  San Bernardino                        Needles Unified   
...             ...             ...                                    ...   
4022      2017-2018     Los Angeles                        Compton Unified   
3981      2017-2018     Los Angeles                     Wilsona Elementary   
10163     2017-2018         Ventura                     Hueneme Elementary   
10164     2017-2018         Ventura                     Hueneme Elementary   
10474     2017-2018          Alpine      Alpine County Office of Education   

                                 School Name                     School Type  \
10395                       Evergreen Middle             Opportunity Schools   
1815                           Adelante High       Continuation High Schools   
4537             Modoc County Juvenile Court          Juvenile Court Schools   
6479   San Benito County Juvenile Hall Court          Juvenile Court Schools   
6736                   Needles Community Day  District Community Day Schools   
...                                      ...                             ...   
4022            Compton Community Day Middle  District Community Day Schools   
3981         Nonpublic, Nonsectarian Schools                             NaN   
10163                        District Office                             NaN   
10164        Nonpublic, Nonsectarian Schools                             NaN   
10474                        District Office                             NaN   

            Educational_Type Low Grade High Grade  1718_Enrollment  \
10395     Opportunity School         7          9                6   
1815     Continuation School         9         12                3   
4537   Juvenile Court School         K         12               11   
6479   Juvenile Court School         K         12                9   
6736    Community Day School         6         12                6   
...                      ...       ...        ...              ...   
4022    Community Day School         6          8                1   
3981                     NaN       NaN        NaN                3   
10163                    NaN         K          8                2   
10164                    NaN       NaN        NaN                1   
10474                    NaN         1         12                0   

       1718_Free_Meal_Count  1718_Free_Meal_Percent  1718_FRPM_Count  \
10395                     6                     1.0                6   
1815                      3                     1.0                3   
4537                     11                     1.0               11   
6479                      9                     1.0                9   
6736                      6                     1.0                6   
...                     ...                     ...              ...   
4022                      0                     0.0                0   
3981                      0                     0.0                0   
10163                     0                     0.0                0   
10164                     0                     0.0                0   
10474                     0                     0.0                0   

       1718_FRPM_Percent  1718_Reduced-Price_Meals_Count  
10395                1.0                               0  
1815                 1.0                               0  
4537                 1.0                               0  
6479                 1.0                               0  
6736                 1.0                               0  
...                  ...                        

It's not that helpful. Let me filter out schools with a Free Meals Percentage of 100%.

In [48]:
data_1718_clean[data_1718_clean['1718_Free_Meal_Percent'] == 1.0]

Academic Year County Name  \
4         2017-2018     Alameda   
11        2017-2018     Alameda   
213       2017-2018     Alameda   
235       2017-2018     Alameda   
419       2017-2018       Butte   
...             ...         ...   
10368     2017-2018        Yolo   
10395     2017-2018        Yolo   
10399     2017-2018        Yolo   
10431     2017-2018        Yuba   
10433     2017-2018        Yuba   

                                           District Name  \
4                     Alameda County Office of Education   
11     California School for the Blind (State Special...   
213                                      Oakland Unified   
235                                      Oakland Unified   
419                     Butte County Office of Education   
...                                                  ...   
10368                    Yolo County Office of Education   
10395                                 Washington Unified   
10399                                 Washington Unified   
10431                    Yuba County Office of Education   
10433                    Yuba County Office of Education   

                                        School Name  \
4                Alameda County Juvenile Hall/Court   
11                  California School for the Blind   
213                            Community Day Middle   
235                   Civicorps Corpsmember Academy   
419                                  Table Mountain   
...                                             ...   
10368                                    Dan Jacobs   
10395                              Evergreen Middle   
10399  West Sacramento School for Independent Study   
10431                   Thomas E. Mathews Community   
10433                              Harry P B Carden   

                          School Type              Educational_Type Low Grade  \
4              Juvenile Court Schools         Juvenile Court School         K   
11              State Special Schools          State Special School         K   
213    District Community Day Schools          Community Day School         6   
235             High Schools (Public)                   Traditional         9   
419            Juvenile Court Schools         Juvenile Court School         K   
...                               ...                           ...       ...   
10368          Juvenile Court Schools         Juvenile Court School         7   
10395             Opportunity Schools            Opportunity School         7   
10399   Alternative Schools of Choice  Alternative School of Choice         K   
10431                County Community       County Community School         7   
10433          Juvenile Court Schools         Juvenile Court School         K   

      High Grade  1718_Enrollment  1718_Free_Meal_Count  \
4             12               77                    77   
11         Adult               40                    40   
213            8               11                    11   
235        Adult               71                    71   
419           12               26                    26   
...          ...              ...                   ...   
10368         12               35                    35   
10395          9                6                     6   
10399         12                4                     4   
10431         12               36                    36   
10433         12               35                    35   

       1718_Free_Meal_Percent  1718_FRPM_Count  1718_FRPM_Percent  \
4                         1.0               77                1.0   
11                        1.0               40                1.0   
213                       1.0               11                1.0   
235                       1.0               71                1.0   
419                       1.0               26                1.0   
...                       ...              ...                ...   
10368                     1.0         

There are 175 schools, but many schools are juvenile court schools. I'd like to see data to exclude juvenile court schools. (Because I'd like to see the change of 1819 and 2122, but juvenile court schools' free meals rate would stay at 100%, so...)

In [49]:
all_free_meals_clean_1718 = data_1718_clean[
    (data_1718_clean['1718_Free_Meal_Percent'] == 1.0) &
    (data_1718_clean['School Type'] != 'Juvenile Court Schools')
].copy()
all_free_meals_clean_1718

Academic Year County Name  \
11        2017-2018     Alameda   
213       2017-2018     Alameda   
235       2017-2018     Alameda   
428       2017-2018       Butte   
479       2017-2018       Butte   
...             ...         ...   
10152     2017-2018     Ventura   
10184     2017-2018     Ventura   
10395     2017-2018        Yolo   
10399     2017-2018        Yolo   
10431     2017-2018        Yuba   

                                           District Name  \
11     California School for the Blind (State Special...   
213                                      Oakland Unified   
235                                      Oakland Unified   
428                                        Chico Unified   
479                             Palermo Union Elementary   
...                                                  ...   
10152                                  Briggs Elementary   
10184                                       Ojai Unified   
10395                                 Washington Unified   
10399                                 Washington Unified   
10431                    Yuba County Office of Education   

                                        School Name  \
11                  California School for the Blind   
213                            Community Day Middle   
235                   Civicorps Corpsmember Academy   
428                              Academy for Change   
479                 Nonpublic, Nonsectarian Schools   
...                                             ...   
10152                               District Office   
10184               Nonpublic, Nonsectarian Schools   
10395                              Evergreen Middle   
10399  West Sacramento School for Independent Study   
10431                   Thomas E. Mathews Community   

                          School Type              Educational_Type Low Grade  \
11              State Special Schools          State Special School         K   
213    District Community Day Schools          Community Day School         6   
235             High Schools (Public)                   Traditional         9   
428    District Community Day Schools          Community Day School         7   
479                               NaN                           NaN       NaN   
...                               ...                           ...       ...   
10152                             NaN                           NaN         K   
10184                             NaN                           NaN       NaN   
10395             Opportunity Schools            Opportunity School         7   
10399   Alternative Schools of Choice  Alternative School of Choice         K   
10431                County Community       County Community School         7   

      High Grade  1718_Enrollment  1718_Free_Meal_Count  \
11         Adult               40                    40   
213            8               11                    11   
235        Adult               71                    71   
428           12               11                    11   
479          NaN                1                     1   
...          ...              ...                   ...   
10152          8                1                     1   
10184        NaN                2                     2   
10395          9                6                     6   
10399         12                4                     4   
10431         12               36                    36   

       1718_Free_Meal_Percent  1718_FRPM_Count  1718_FRPM_Percent  \
11                        1.0               40                1.0   
213                       1.0               11                1.0   
235                       1.0               71                1.0   
428                       1.0               11                1.0   
479                       1.0                1                1.0   
...                       ...              ...                ...   
10152                     1.0         

Now we can see there're 110 schools for the year 17-18. (By the way, I'm surprised by the number of Juvenile Court Schools...)

Then how about other years?

In [50]:
all_free_meals_clean_1819 = data_1819_clean[
    (data_1819_clean['1819_Free_Meal_Percent'] == 1.0) &
    (data_1819_clean['School Type'] != 'Juvenile Court Schools')
].copy()
all_free_meals_clean_1819

Academic Year County Name  \
12        2018-2019     Alameda   
13        2018-2019     Alameda   
503       2018-2019       Butte   
504       2018-2019       Butte   
509       2018-2019       Butte   
...             ...         ...   
9967      2018-2019     Trinity   
9984      2018-2019      Tulare   
10010     2018-2019      Tulare   
10183     2018-2019    Tuolumne   
10444     2018-2019        Yolo   

                                           District Name  \
12     California School for the Blind (State Special...   
13     California School for the Deaf-Fremont (State ...   
503                          Thermalito Union Elementary   
504                          Thermalito Union Elementary   
509                          Thermalito Union Elementary   
...                                                  ...   
9967                             Mountain Valley Unified   
9984                                     Alpaugh Unified   
10010                             Hot Springs Elementary   
10183                             Summerville Union High   
10444                                 Washington Unified   

                                        School Name  \
12                  California School for the Blind   
13           California School for the Deaf-Fremont   
503                 Nonpublic, Nonsectarian Schools   
504                           Pioneer Community Day   
509                          Heritage Community Day   
...                                             ...   
9967                                    Valley High   
9984                         Tule Continuation High   
10010                        Hot Springs Elementary   
10183                             Cold Springs High   
10444  West Sacramento School for Independent Study   

                                     School Type  \
12                         State Special Schools   
13                         State Special Schools   
503                                          NaN   
504               District Community Day Schools   
509               District Community Day Schools   
...                                          ...   
9967                   Continuation High Schools   
9984                   Continuation High Schools   
10010  Elemen Schools In 1 School Dist. (Public)   
10183                      High Schools (Public)   
10444              Alternative Schools of Choice   

                   Educational_Type Low Grade High Grade  1819_Enrollment  \
12             State Special School         K      Adult               66   
13             State Special School         P         12              372   
503                             NaN       NaN        NaN                3   
504            Community Day School         1          6                6   
509            Community Day School         4          8                9   
...                             ...       ...        ...              ...   
9967            Continuation School         9         12                6   
9984            Continuation School        10         12                1   
10010                   Traditional         K          8               19   
10183                   Traditional         9         12                2   
10444  Alternative School of Choice         K         12                1   

       1819_Free_Meal_Count  1819_Free_Meal_Percent  1819_FRPM_Count  \
12                       66                     1.0               66   
13                      372                     1.0              372   
503                       3                     1.0                3   
504                       6                     1.0                6   
509                       9                     1.0                9   
...                     ...                     ...              ...   
9967                      6                     1.0                6   
9984                      1                     1.0                1   
10010

In [51]:
all_free_meals_clean_1920 = data_1920_clean[
    (data_1920_clean['1920_Free_Meal_Percent'] == 1.0) &
    (data_1920_clean['School Type'] != 'Juvenile Court Schools')
].copy()
all_free_meals_clean_1920

Academic Year   County Name  \
14        2019-2020       Alameda   
491       2019-2020         Butte   
658       2019-2020  Contra Costa   
847       2019-2020     Del Norte   
893       2019-2020     El Dorado   
...             ...           ...   
10246     2019-2020      Tuolumne   
10289     2019-2020       Ventura   
10337     2019-2020       Ventura   
10528     2019-2020          Yolo   
10580     2019-2020          Yuba   

                                           District Name  \
14     California School for the Blind (State Special...   
491                          Thermalito Union Elementary   
658                                   Mt. Diablo Unified   
847                 Del Norte County Office of Education   
893                          Gold Trail Union Elementary   
...                                                  ...   
10246                             Summerville Union High   
10289                                         Ocean View   
10337                                    Pleasant Valley   
10528                             Woodland Joint Unified   
10580                                          Wheatland   

                               School Name                     School Type  \
14         California School for the Blind           State Special Schools   
491                  Pioneer Community Day  District Community Day Schools   
658            Gateway High (Continuation)       Continuation High Schools   
847                    Del Norte Community                County Community   
893        Nonpublic, Nonsectarian Schools                             NaN   
...                                    ...                             ...   
10246                    Cold Springs High           High Schools (Public)   
10289                      District Office                             NaN   
10337                      District Office                             NaN   
10528  Woodland Middle Grade Community Day  District Community Day Schools   
10580      Nonpublic, Nonsectarian Schools                             NaN   

              Educational_Type Low Grade High Grade  1920_Enrollment  \
14        State Special School         K      Adult               64   
491       Community Day School         1          6                6   
658        Continuation School         9         12                1   
847    County Community School         K         12               26   
893                        NaN       NaN        NaN                1   
...                        ...       ...        ...              ...   
10246              Traditional         9         12                1   
10289                      NaN         P          8                1   
10337                      NaN         P         12                1   
10528     Community Day School         7          8                3   
10580                      NaN       NaN        NaN                1   

       1920_Free_Meal_Count  1920_Free_Meal_Percent  1920_FRPM_Count  \
14                       64                     1.0               64   
491                       6                     1.0                6   
658                       1                     1.0                1   
847                      26                     1.0               26   
893                       1                     1.0                1   
...                     ...                     ...              ...   
10246                     1                     1.0                1   
10289                     1                     1.0                1   
10337                     1                     1.0                1   
10528                     3                     1.0                3   
10580                     1                     1.0                1   

       1920_FRPM_Percent  1920_Reduced-Price_Meals_Count  
14                   1.0                               0  
491                  1.0                               

In [52]:
all_free_meals_clean_2021 = data_2021_clean[
    (data_2021_clean['2021_Free_Meal_Percent'] == 1.0) &
    (data_2021_clean['School Type'] != 'Juvenile Court Schools')
].copy()
all_free_meals_clean_2021

Academic Year County Name  \
13        2020-2021     Alameda   
230       2020-2021     Alameda   
266       2020-2021     Alameda   
470       2020-2021       Butte   
492       2020-2021       Butte   
...             ...         ...   
10277     2020-2021     Ventura   
10304     2020-2021     Ventura   
10327     2020-2021     Ventura   
10432     2020-2021     Ventura   
10527     2020-2021        Yuba   

                                           District Name  \
13     California School for the Blind (State Special...   
230                                      Oakland Unified   
266                                      Oakland Unified   
470                                  Oroville Union High   
492                             Pioneer Union Elementary   
...                                                  ...   
10277                                             Oxnard   
10304                                    Pleasant Valley   
10327                                Simi Valley Unified   
10432                                Santa Paula Unified   
10527                           Marysville Joint Unified   

                           School Name  \
13     California School for the Blind   
230      Civicorps Corpsmember Academy   
266       Infant and Preschool Program   
470        Oroville High Community Day   
492             Berry Creek Elementary   
...                                ...   
10277  Nonpublic, Nonsectarian Schools   
10304  Nonpublic, Nonsectarian Schools   
10327                  District Office   
10432  Nonpublic, Nonsectarian Schools   
10527         Marysville Community Day   

                                     School Type      Educational_Type  \
13                         State Special Schools  State Special School   
230                        High Schools (Public)           Traditional   
266                                    Preschool           Traditional   
470               District Community Day Schools  Community Day School   
492    Elemen Schools In 1 School Dist. (Public)           Traditional   
...                                          ...                   ...   
10277                                        NaN                   NaN   
10304                                        NaN                   NaN   
10327                                        NaN                   NaN   
10432                                        NaN                   NaN   
10527             District Community Day Schools  Community Day School   

      Low Grade High Grade  2021_Enrollment  2021_Free_Meal_Count  \
13            K      Adult               58                    58   
230           9      Adult               54                    54   
266           P          P                5                     5   
470           9         12                9                     9   
492           K          8               52                    52   
...         ...        ...              ...                   ...   
10277       NaN        NaN                5                     5   
10304       NaN        NaN                1                     1   
10327         P      Adult                1                     1   
10432       NaN        NaN                1                     1   
10527         7         12               19                    19   

       2021_Free_Meal_Percent  2021_FRPM_Count  2021_FRPM_Percent  \
13                        1.0               58                1.0   
230                       1.0               54                1.0   
266                       1.0                5                1.0   
470                       1.0                9                1.0   
492                       1.0               52                1.0   
...                       ...              ...                ...   
10277                     1.0                5                1.0   
10304                     1.0                1                1.0   
10327               

In [53]:
all_free_meals_clean_2122 = data_2122_clean[
    (data_2122_clean['2122_Free_Meal_Percent'] == 1.0) &
    (data_2122_clean['School Type'] != 'Juvenile Court Schools')
].copy()
all_free_meals_clean_2122

Academic Year County Name  \
14        2021-2022     Alameda   
15        2021-2022     Alameda   
229       2021-2022     Alameda   
466       2021-2022       Butte   
468       2021-2022       Butte   
...             ...         ...   
10319     2021-2022     Ventura   
10441     2021-2022     Ventura   
10473     2021-2022        Yolo   
10514     2021-2022        Yuba   
10553     2021-2022        Yuba   

                                           District Name  \
14     California School for the Blind (State Special...   
15     California School for the Deaf-Fremont (State ...   
229                                      Oakland Unified   
466                             Palermo Union Elementary   
468                             Palermo Union Elementary   
...                                                  ...   
10319                                     Rio Elementary   
10441                                Santa Paula Unified   
10473                                 Washington Unified   
10514                    Yuba County Office of Education   
10553                               Wheatland Union High   

                                  School Name                     School Type  \
14            California School for the Blind           State Special Schools   
15     California School for the Deaf-Fremont           State Special Schools   
229                             Community Day  District Community Day Schools   
466                                    Honcut     Elementary Schools (Public)   
468         Palermo Union Community Day (K-8)  District Community Day Schools   
...                                       ...                             ...   
10319         Nonpublic, Nonsectarian Schools                             NaN   
10441         Nonpublic, Nonsectarian Schools                             NaN   
10473                        Evergreen Middle             Opportunity Schools   
10514             Thomas E. Mathews Community                County Community   
10553                         District Office                             NaN   

              Educational_Type Low Grade High Grade  2122_Enrollment  \
14        State Special School         K      Adult               62   
15        State Special School         P         12              318   
229       Community Day School         6         12                8   
466                Traditional         K          3               14   
468       Community Day School         K          8                1   
...                        ...       ...        ...              ...   
10319                      NaN       NaN        NaN                1   
10441                      NaN       NaN        NaN                1   
10473       Opportunity School         7          9                6   
10514  County Community School         7         12               49   
10553                      NaN         7         12                3   

       2122_Free_Meal_Count  2122_Free_Meal_Percent  2122_FRPM_Count  \
14                       62                     1.0               62   
15                      318                     1.0              318   
229                       8                     1.0                8   
466                      14                     1.0               14   
468                       1                     1.0                1   
...                     ...                     ...              ...   
10319                     1                     1.0                1   
10441                     1                     1.0                1   
10473                     6                     1.0                6   
10514                    49                     1.0               49   
10553                     3                     1.0                3   

       2122_FRPM_Percent  2122_Reduced-Price_Meals_Count  
14                   1.0                               0  
15                   1.0                   

This is interesting, the data from 1718 to 2122 is: 110, 118, 104, 121, 110. Also noted that the enrollment in 2122 is far lower than 1718, so it's safe to say:

4. After the pandemic, proportion of schools with 100% free meals is greater than before the pandemic. 